src/ui/configuration/ApmSoftwareConfig.cc

```diff
@@ -163,7 +183,12 @@ void ApmSoftwareConfig::apmParamNetworkReplyFinished()
 
     } else {
         QByteArray apmpdef = reply->readAll();
-        m_apmPdefFilename = QDir(QGC::appDataDirectory()).filePath("apm.pdef.xml");
+
+        QDir autopilotdir(QGC::shareDirectory() + "/files/" + "ardupilotmega");
```

Comment:
Why did you change the data path from `appDataDirectory()` to `shareDirectory()`? On my Ubuntu 16.04 `shareDirectory()` delivers the path to the binary of apm planner - on some systems the User may not have the rights to write to that folder. Moreover the path is extended by `/files/ardupilotmega` but not checked if valid, if it does not exist it will **not** be automatically created and therefore the `file.open(QIODevice::ReadWrite | QIODevice::Truncate)` will always fail.
I would prefer using the old path which is in `home/apmplanner2` and perhaps we should create a subfolder?!